web/src/components/index/UserTasks.tsx

```diff
@@ -1,17 +1,51 @@
 import { useAuth } from "@/hooks/useAuth";
 import LoginFallback from "@/components/ui/LoginFallback";
+import Button from "../ui/Button";
+import { FormEvent, useRef } from "react";
+import styles from "./UserTasks.module.css";
+import { postProfile } from "@/openapi";
 
 export default function UserTasks() {
   const { loading, isAuthed, userData } = useAuth();
+  const dialogRef = useRef<HTMLDialogElement>(null);
 
+  const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const inputElement = event.currentTarget.elements.namedItem("chosenName") as HTMLInputElement;
+    const chosenName = inputElement.value;
+    const dialogElement = dialogRef.current;
+
+    try {
+      const response = await postProfile({
+        body: { chosen_name: chosenName },
+      });
+      if (!response.response.ok) {
+        throw new Error(`HTTP error! status: ${response.response.status}`);
+      } else {
+        dialogElement!.close();
+      }
+    } catch (error) {
+      console.error("There was a problem submitting your request:", error);
+    }
+  };
   if (loading) return null;
 
   if (!isAuthed) return <LoginFallback />;
 
   return (
     <div>
+      {!userData!.chosen_name && (
+        <dialog open ref={dialogRef} id={styles.chosenName} data-id="chosenName">
+          <form onSubmit={handleSubmit}>
+            Please enter your name as you would choose to have it appear on forms related to our services.
+            <input type="text" name="chosenName"></input>
+            <Button type="submit">Submit</Button>
+          </form>
+        </dialog>
+      )}
       <p>
-        Username&nbsp;{userData!.username}. Roles:&nbsp;
+        Name: {userData!.chosen_name}. Username&nbsp;{userData!.username}. Roles:&nbsp;
```

Comment:
this doesn't get updated. perhaps should be `useState`?

Reply:
should I be updating the auth context for this or am I overthinking it

Reply:
Have added a state just for chosen name, was previously thinking for userData